### PR TITLE
Fix bug causing log-likelihoods to be computed incorrectly.

### PR DESCRIPTION
**Pull request by ashelkovnykov**
- Logistic and Poisson log-likelhoods computed incorrectly
- Average computed incorrectly, causing stochastic behaviour
- The above resulted in a mismatch between the metrics reported for Poisson regression, and the ones used to choose the best model (both were wrong)

### CODE REVIEW

File: photon-diagnostics/src/main/scala/com/linkedin/photon/ml/Evaluation.scala [hunk 1]
```diff
@@ -129,49 +130,46 @@ object Evaluation extends Logging {
...
+  private def averageLogLikelihoodRDD(logLikelihoods: RDD[(Double, Int)]): Double = {
+
+    val (logLikelihood, count) = logLikelihoods.reduce { (a, b) =>
+      val (logLikelihoodA, countA) = a
+      val (logLikelihoodB, countB) = b
+
+      (logLikelihoodA + logLikelihoodB, countA + countB)
+    }
+
+    logLikelihood - math.log(count)
```
Comment (ankans) [inline photon-diagnostics/src/main/scala/com/linkedin/photon/ml/Evaluation.scala]: Where is this formula coming from?

File: photon-diagnostics/src/main/scala/com/linkedin/photon/ml/Evaluation.scala [hunk 2]
```diff
@@ -129,49 +130,46 @@ object Evaluation extends Logging {
...
-        count += 1
-        mean += (x - mean) / count
-      })
-      Seq((count, mean)).iterator
-    }).reduce((a, b) => {
-      // Aggregate per-partition means
-      val (countA, meanA) = a
-      val (countB, meanB) = b
-      val newCount = countA + countB
-      val newMean = meanA + (meanB * countB) / newCount
```
Comment (ankans) [inline photon-diagnostics/src/main/scala/com/linkedin/photon/ml/Evaluation.scala]: Shouldn't this have been meanA*countA +.... ?

Comment (ashelkovnykov) [reply photon-diagnostics/src/main/scala/com/linkedin/photon/ml/Evaluation.scala]: Yeah

File: photon-diagnostics/src/main/scala/com/linkedin/photon/ml/Evaluation.scala (hunk 1 above)
Comment (ashelkovnykov) [reply photon-diagnostics/src/main/scala/com/linkedin/photon/ml/Evaluation.scala]: From a misunderstanding I had - changing it

**Pull request merged**